presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestDistributedQueries.java

```diff
@@ -440,6 +440,12 @@ public void testDelete()
                         "WHERE (orderkey IN (SELECT CASE WHEN orderkey % 3 = 0 THEN NULL ELSE orderkey END FROM lineitem)) IS NOT NULL\n");
 
         assertUpdate("DROP TABLE test_delete");
+
+        // delete using a scalar subquery
+
+        assertUpdate("CREATE TABLE test_delete AS SELECT * FROM orders", "SELECT count(*) FROM orders");
+        assertUpdate("DELETE FROM test_delete WHERE orderkey = (SELECT orderkey FROM orders ORDER BY orderkey LIMIT 1)", 0);
```

Comment:
Does `0` mean that nothing was deleted? If so can you change this test to actually delete some rows. Additionally can check if correct rows were removed (like `assertQuery` checks the selection).